app/models/collection_item.rb

```diff
@@ -342,4 +342,13 @@ def notify_of_unrevealed_or_anonymous
       ).deliver_after_commit
     end
   end
+
+  # reindex collection after creation, deletion, and approval_status update
+  after_create :reindex_collection
+  after_destroy :reindex_collection
+  after_update :reindex_collection, if: Proc.new { |c| c.saved_change_to_user_approval_status? || c.saved_change_to_collection_approval_status? }
```

Comment:
Similar to the challenge classes, you can provide a list of attributes to check (`saved_changes.keys & pertinent_attributes`), and you should use an `after_commit` callback.